### DISCUSSION TITLE
Whether it is possible to use a software timer or a hardware timer to provide the tick?

### DISCUSSION THREAD

**Discussion by AnemosCiel** [Lora-net/LoRaMac-node#1232] [Q&A]
Since MCU does not have RTC function, additional RTC chip needs to be introduced. I am considering whether software timer or hardware timer can be used to provide a tick without RTC. If so, how can I reconstruct RTC.c? Which functions are required?

**Comment by mluis1** [comment]
Any timer source can be used. 
We used the RTC in this project because it was the only clock source that did not stop when setting the MCU in sleep mode (STM32 platforms).
